dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -1215,39 +1266,143 @@ public void Test_Create_Offending_Secrets_Missing_Site_Then_Export_Them_Then_Imp
+            //Save it
+            api.saveSecrets(secrets, site, admin);
+            final Optional<AppSecrets> siteSecretsOptional = api.getSecrets(appKey, site, admin);
 
-        //Now we're gonna create an inconsistency removing the app descriptor and then try to import the secret.
-        final HostAPI hostAPI = APILocator.getHostAPI();
-        hostAPI.archive(site, admin, false);
-        hostAPI.delete(site, admin, false);
+            api.saveSecrets(secrets, systemHost, admin);
+            final Optional<AppSecrets> systemHostSecretsOptional = api
+                    .getSecrets(appKey, systemHost, admin);
+
+            //AES only supports security Key of sizes of 16, 24 or 32 bytes.
+            final String password = RandomStringUtils.randomAlphanumeric(32);
+            final Key securityKey = AppsUtil.generateKey(password);
+
+            //Now that we have a valid key lets dump all our secrets.
+            final Path exportSecretsFile = api.exportSecrets(securityKey, true, null, admin);
+            assertTrue(exportSecretsFile.toFile().exists());
+
+            assertTrue(exportSecretsFile.toFile().exists());
+
+            //Remove the secret we dumped we can re import it.
+            api.deleteSecrets(appKey, site, admin);
+            api.deleteSecrets(appKey, systemHost, admin);
+
+            //and finally import it
+            api.importSecretsAndSave(exportSecretsFile, securityKey, admin);
 
-        //and finally import it
-        api.importSecretsAndSave(exportSecretsFile, securityKey, admin);
+            //verify
+            final Optional<AppSecrets> secretsOptionalPostImport1 = api.getSecrets(appKey, site, admin);
+            assertTrue(secretsOptionalPostImport1.isPresent());
+            final AppSecrets restoredSecrets1 = secretsOptionalPostImport1.get();
+
+            final Optional<AppSecrets> secretsOptionalPostImport2 = api.getSecrets(appKey, systemHost, admin);
+            assertTrue(secretsOptionalPostImport2.isPresent());
+            final AppSecrets restoredSecrets2 = secretsOptionalPostImport2.get();
+
+            assertEquals(restoredSecrets1.getKey(),appKey);
+            assertEquals(restoredSecrets2.getKey(),appKey);
+
+            assertTrue(restoredSecrets1.getSecrets().containsKey("p1"));
+            assertTrue(restoredSecrets2.getSecrets().containsKey("p2"));
+
+        }finally {
+           file.delete();
+        }
+
+    }
+
+    /**
+     * Method to test {@link ExportSecretForm#toString()}
+     * Given scenario: We feed the form with null and non null values
+     * Expected results: We expect the form to behave NPE Free when calling toString
+     */
+    @Test
+    public void Test_NPE_Free_ToString(){
+        final ExportSecretForm formAllNull = new ExportSecretForm(null, false, null);
+        assertNotNull(formAllNull.toString());
 
+        final ExportSecretForm formNotAllNull = new ExportSecretForm(null, false, ImmutableMap.of("1", ImmutableSet.of("1","2")));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'ImmutableMap.of' due to existing static import 'com.google.common.collect.ImmutableMap.of'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392590)